testing/src/test/java/io/stargate/it/http/RestApiTableTokenAuthTest.java

```diff
@@ -0,0 +1,213 @@
+public class RestApiTableTokenAuthTest extends BaseOsgiIntegrationTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(RestApiTableTokenAuthTest.class);
+
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private final String keyspaceName = "table_token_test";
+  private final String tableName = "tbl_test";
+  private final String readOnlyUsername = "read_only_user";
+  private final String readOnlyPassword = "read_only_user";
+
+  private String host;
+
+  @SuppressWarnings("unused") // referenced in @StargateSpec
+  public static void buildParameters(StargateParameters.Builder builder) {
+    builder.enableAuth(true);
+    builder.putSystemProperties("stargate.auth_id", "AuthTableBasedService");
+  }
+
+  @BeforeEach
+  public void setup(StargateConnectionInfo cluster) {
+    host = "http://" + cluster.seedAddress();
+
+    objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+  }
+
+  @Test
+  public void getRowsV2() throws IOException {
+    String authToken = generateToken(readOnlyUsername, readOnlyPassword);
+
+    String body =
+        RestUtils.get(
+            authToken,
+            String.format("%s:8082/v2/keyspaces/%s/%s/%s", host, keyspaceName, tableName, "a"),
+            HttpStatus.SC_OK);
+
+    GetResponseWrapper getResponseWrapper = objectMapper.readValue(body, GetResponseWrapper.class);
+    List<Map<String, Object>> data =
+        objectMapper.convertValue(
+            getResponseWrapper.getData(), new TypeReference<List<Map<String, Object>>>() {});
+
+    for (Map<String, Object> row : data) {
+      assertThat(row.get("key")).isNotNull();
+      assertThat(row.get("value")).isNotNull();
+    }
+  }
+
+  @Test
+  public void getRowsV2NotAuthorized() throws IOException {
+    String keyspace = "ks_unauthorized";
+    String table = "tbl1";
+
+    String createKeyspaceRequest = String.format("{\"name\": \"%s\", \"replicas\": 1}", keyspace);
+
+    String authToken = generateToken("cassandra", "cassandra");
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces", host),
+        createKeyspaceRequest,
+        HttpStatus.SC_CREATED);
+
+    TableAdd tableAdd = new TableAdd();
+    tableAdd.setName(table);
+
+    List<ColumnDefinition> columnDefinitions = new ArrayList<>();
+
+    columnDefinitions.add(new ColumnDefinition("k", "text"));
+    columnDefinitions.add(new ColumnDefinition("v", "text"));
+
+    tableAdd.setColumnDefinitions(columnDefinitions);
+
+    PrimaryKey primaryKey = new PrimaryKey();
+    primaryKey.setPartitionKey(Collections.singletonList("k"));
+    tableAdd.setPrimaryKey(primaryKey);
+
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/tables", host, keyspace),
+        objectMapper.writeValueAsString(tableAdd),
+        HttpStatus.SC_CREATED);
+
+    try {
+      // Need to sleep to allow propagation of permissions
+      Thread.sleep(5000L);
```

Comment:
can we use `Awaiatality` (https://github.com/awaitility/awaitility) with `await.until()`?

Reply:
Actually this is no longer needed. Forgot to remove after debugging.